support-services/src/main/scala/com/gu/zuora/ZuoraGiftService.scala

```diff
@@ -44,8 +50,73 @@ class ZuoraGiftService(val config: ZuoraConfig, client: FutureHttpClient)(implic
     gifteeIdentityId: String,
     giftRedemptionDate: LocalDate,
     newTermLength: Int
-  ): Future[UpdateRedemptionDataResponse] = {
+  ): Future[ZuoraSuccessOrFailureResponse] = {
     val requestData = UpdateRedemptionDataRequest(requestId, gifteeIdentityId, giftRedemptionDate, newTermLength, Day)
-    putJson[UpdateRedemptionDataResponse](s"subscriptions/${subscriptionId}", requestData.asJson, authHeaders)
+    val response = putJson[ZuoraSuccessOrFailureResponse](s"subscriptions/${subscriptionId}", requestData.asJson, authHeaders)
+    response.transform{
+        case Success(successfulResponse) if successfulResponse.success =>
+          SafeLogger.info(s"Successfully updated gifteeIdentityId on Digital Subscription gift for user $gifteeIdentityId")
+          Success(successfulResponse)
+        case Success(failedResponse) =>
+          // Treat a ZuoraResponse where success = false as a failure
+          val responseMessage = failedResponse.errorMessage.getOrElse("unknown")
+          val errorMessage = s"Failed to update gifteeIdentityId on Digital Subscription gift for user $gifteeIdentityId. Error was ${responseMessage}"
+          Failure(new RuntimeException(errorMessage))
+        case Failure(originalError) =>
+          val errorMessage = s"Failed to update gifteeIdentityId on Digital Subscription gift for user $gifteeIdentityId."
+          Failure(new RuntimeException(errorMessage, originalError))
+      }
```

Comment:
it's very verbose but this will make it clear what happened when things go wrong 👏